src/app/Dashboard/DraggableRef.tsx

```diff
@@ -0,0 +1,139 @@
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { dashboardCardConfigResizeCardIntent } from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
+import { gridSpans } from '@patternfly/react-core';
+import _ from 'lodash';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { DashboardCardContext } from './ResizableCard';
+
+export interface DraggableRefProps {
+  dashboardIdx: number;
+  minimumSpan: gridSpans;
+}
+
+function normalizeAsGridSpans(val: number, min: number, max: number, a: number, b: number): gridSpans {
+  if (val < min) val = min;
+  else if (val > max) val = max;
+  const ans = Math.round((b - a) * ((val - min) / (max - min)) + a);
+  return _.clamp(ans, a, b) as gridSpans;
+}
+
+export const DraggableRef: React.FunctionComponent<DraggableRefProps> = (props) => {
+  const dispatch = useDispatch();
+
+  const cardRef = React.useContext(DashboardCardContext);
+  const draggableRef = React.useRef<HTMLDivElement>(null);
+  const isResizing = React.useRef<boolean>(false);
+  const setInitialVals = React.useRef<boolean>(true);
+
+  const SMALLEST_CARD_WIDTH = 126;
+
+  const handleOnMouseDown = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    e.stopPropagation();
+    e.preventDefault();
+    document.body.style.setProperty('cursor', 'col-resize');
+    document.addEventListener('mousemove', callbackMouseMove);
+    document.addEventListener('mouseup', callbackMouseUp);
+    isResizing.current = true;
+    setInitialVals.current = true;
+  };
+
+  const handleMouseMove = (e: MouseEvent) => {
+    const mousePos = e.clientX;
+    e.stopPropagation();
+    if (!isResizing.current) {
+      return;
+    }
+    if (cardRef.current && window.visualViewport) {
+      cardRef.current.scrollIntoView({ behavior: 'smooth', block: 'center', inline: 'end' });
+      const cardRect = cardRef.current.getBoundingClientRect();
+      const cardLeft = cardRect.left + SMALLEST_CARD_WIDTH * props.minimumSpan;
```

Comment:
Seems like we are looking at the right-side of the rectangle here? Should be renamed?